components/telemetry-operator/controllers/telemetry/logpipeline_controller.go

```diff
@@ -186,6 +179,38 @@ func shouldRetryOn(err error) bool {
 		!errors.IsNotFound(err)
 }
 
+func getError(crUpdateError, fbRestartError error) error {
```

Comment:
Let's make this the following: `func getFirstError(err ...error) error`
The name describes better what is actually happening and you could just return the first list element (or nil). Maybe let's move this to a new `error_helpers` file together with `shouldRetryOn` to do not use code from a different controller.